### ISSUE TITLE
user can archive polls

### ISSUE THREAD

**Issue by mensch72** [pik-gane/vodle#83]
Into a special "archived" section in "my polls".

**Comment by ARSHIL1804** [comment]
Is this still open I would like to work on this

**Comment by mensch72** [comment]
Hi @ARSHIL1804 , thank you for your offer! It is still open, I'll assign you. Do you need more information?

**Comment by ARSHIL1804** [comment]
what should be the behavior of archived polls? Will they be stored in remain in poll db?

**Comment by mensch72** [comment]
I suggest the following:
- Only polls that have ended already can be archived
- To allow the user to archive a poll, the poll page shows a checkbox labelled "archive" at its bottom, which is initially unchecked. The value of this checkbox is stored in an attribute in the `Poll` class in `poll.service.ts` similar to the attribute `have_seen_results`. 
- Archived polls are shows on the "my polls" page in a separate category below "running", "ended", and "drafts".
- Data of archived polls is kept on the device even if it gets deleted on the server (which happens automatically after 30 days). For this, code in `data.service.ts` needs to be changed so that `delete_poll_data` is not triggered for archived polls.
- When unchecking the "archive" button for a poll that is 30 days over its due date, a warning should be shown that this will likely lead to that poll's data being deleted.
Does that help?

**Comment by ARSHIL1804** [comment]
Yeah, Now I got the use of the archive. I was thinking in a whole other direction Thanks @mensch72

**Comment by mensch72** [comment]
Sorry, I just saw this question by you. I would say that on the "mypolls" (!) page, where all polls are listed, archived polls should only show up in the additional section "archived" below "ended", and there should not be any checkboxes on the mypolls page. To control whether an ended poll is archived or not, there should instead be a a checkbox on that poll's "poll" page, at the very bottom. You can use a standard ionic checkbox for this, similar to those used on the "settings" page.

**Comment by ARSHIL1804** [comment]
Thanks @mensch72

**Comment by ARSHIL1804** [comment]
@mensch72  When you said that the archived poll should kept on the device, do you mean local storage? 
Also currently,  how ended poll data got deleted after 30 days
?

**Comment by mensch72** [comment]
Yes, exactly, in the local storage.

Currently, it might get deleted when the server deletes data from the database after a certain time, and that deletion then gets synced to local storage as part of the pouchdb syncing process in data.service.ts .

**Comment by ARSHIL1804** [comment]
@mensch72  I have raised my #254 .
I have made changes to not delete the poll if it is archived as per my little couch db knowledge. 
Though I don't really understand difference between local_only_user_DB,local_synced_user_d,remote_user_db,local_poll_dbs,remote_poll_dbs.

Please suggest me changes for the same. 
Thank you.
